### DISCUSSION TITLE
Generative art

### DISCUSSION THREAD

**Discussion by yangwao** [kodadot/nft-gallery#3335] [Hackathon]
We've already started researching various experiments with on-chain SVGs (check subsalt channel on Discord) where the original goal was to drive p5.js through some sort of sandbox where people can mint their code and particular generative art would be rendered on client based on conditional rendering. 

Apart from that, there were few creative hacks we could leverage use, I've started researching how to drive other people easily minting and discover other machine learning methods (dalle mini, dalle mega) to drive more various mints.

For experimental minter would be to **hack** API for replicate.com so users could use it right away from KodaDot, like starting with `kuprel/min-dalle` model.

### Actionable links 
- https://github.com/nicholascelestin/replicate-js
- https://replicate.com/docs/api/getting-started-public
- https://replicate.com/kuprel/min-dalle
-  https://replicate.com/api#pricing

### Models 
- https://replicate.com/laion-ai/puck
- https://replicate.com/jack000/glid-3-xl
- https://replicate.com/netease-gameai/spatchgan-selfie2anime
- https://replicate.com/collections/ml-makeovers
- other models https://replicate.com/explore

**Comment by jkdizajn** [comment]
Would it be very technically challenging to implement an NFT generator in KodaDot (Basilisk or Moonbeam)?
I'm thinking of those collections that have even 1000-10000 NFTs. 

I'll give an example:

As an artist I have created a project that has:
- 10 different backgrounds
- 10 bodies
- 10 heads
- 10 eyes
- 10 different legs

I would like to upload all of this to the KodaDot server and mix it up to create a collection of 10000 original NFTs
Of course there are external generators for this, but they are either for a fee or for some other hidden service, having it directly in KodaDot would be a big attraction for new artists. 

As an example of such a service: 
https://www.bueno.art/
https://speedboat.studio/
https://github.com/NotLuksus/nft-art-generator
